### PR TITLE
fix for some minor inconsistencies

### PR DESCRIPTION
**Pull request by MartinWeindel**
**What this PR does / why we need it**:
Several minor issues in sample machine_server.go have been fixed:
- document in sample code to return NOT_FOUND in GetMachineStatus if machine is not existing
- sample code for decoding in GetVolumeIDs method
- no logging of big request data
- typos in logging messages

### CODE REVIEW

File: pkg/sampleprovider/machine_server.go [hunk 1]
```diff
@@ -121,7 +128,7 @@ func (ms *MachinePlugin) GetMachineStatus(ctx context.Context, req *cmi.GetMachi
 //
 func (ms *MachinePlugin) ListMachines(ctx context.Context, req *cmi.ListMachinesRequest) (*cmi.ListMachinesResponse, error) {
 	// Log messages to track start of request
-	glog.V(2).Infof("List machines request has been recieved for %q", req.ProviderSpec)
+	glog.V(2).Infof("List machines request has been received")
```
Comment (prashanth26) [inline pkg/sampleprovider/machine_server.go:131]: Do you think we shouldn't print the `providerSpec` here? I ask because even I am not sure about it.

**Review by prashanth26 — COMMENTED** [review]
Hi @MartinWeindel ,

Thanks for the PR. Really appreciate it.

File: pkg/sampleprovider/machine_server.go (hunk 1 above)
Comment (MartinWeindel) [reply pkg/sampleprovider/machine_server.go:131]: It produces quite a lot of logging. What do you think about logging it only at a verbose level of 4 (see additional commit https://github.com/gardener/machine-controller-manager-provider-sampleprovider/pull/15/commits/f72c6b8fd09c5586112f12fabdd2202117ffe439)?

Comment (prashanth26) [reply pkg/sampleprovider/machine_server.go:131]: Yes, I think this should be better.

**Review by prashanth26 — APPROVED** [review]

**Pull request merged**